root_orchestrator/system-manager-python/tests/services_test.py

```diff
@@ -1,278 +1,268 @@
 import copy
-import importlib
+import os
 import sys
-import unittest
-from unittest.mock import ANY, MagicMock
+from unittest.mock import ANY, MagicMock, Mock, patch
 
-import mongomock as mongomock
-import pymongo
-from ext_requests import apps_db, mongodb_client
-from tests.utils import get_full_random_sla_app
+import pytest
+from rasclient import app_operations, job_operations
+from testcontainers.compose import DockerCompose
+from tests.utils import get_full_random_sla_app  # noqa: E402
 
-sys.modules["ext_requests.net_plugin_requests"] = unittest.mock.Mock()
+sys.modules["ext_requests.net_plugin_requests"] = Mock()
 net_plugin = sys.modules["ext_requests.net_plugin_requests"]
 net_plugin.net_inform_service_deploy = MagicMock()
 
-# Note:
-# We need to run the code above before importing the following, otherwise errors occur.
-# We could simply use normal imports and it will work, but this violates the PEP8 code style.
-sm = importlib.import_module("services.service_management")
-create_services_of_app = sm.create_services_of_app
-delete_service = sm.delete_service
-generate_db_structure = sm.generate_db_structure
-get_all_services = sm.get_all_services
-get_service = sm.get_service
-update_service = sm.update_service
-user_services = sm.user_services
+# we ignore E402 because we need to import the service_management module after the mock
+from services.service_management import (  # noqa: E402
+    create_services_of_app,
+    delete_service,
+    generate_db_structure,
+    get_all_services,
+    get_service,
+    update_service,
+    user_services,
+)
 
+current_dir = os.path.dirname(os.path.abspath(__file__))
 
-@mongomock.patch(servers=(("localhost", 10007),))
-def test_create_service_with_app():
-    # SETUP
-    mockdb()
 
-    sla = get_full_random_sla_app()
-    db_app_mock = copy.deepcopy(sla)
+@pytest.fixture(scope="session")
+def resource_abstractor():
+    with DockerCompose(current_dir) as compose:
+        service_uri = "http://localhost:21011"  # TODO: have it fetched from compose object.
```

Comment:
Could you do this already?

Reply:
I believe I can. I'll fix it.

Reply:
Actually looking at it again, it seems the best thing i can do is do this:
```py
service_uri = (
            f"http://{compose.get_service_host('resource_abstractor', 21011)}"
            f":{compose.get_service_port('resource_abstractor', 20007)}"
        )
```
In which case I still need to specify the port. It is slightly better because it returns the mapped port - in case the mapping is not the same i.e 21011 doesn't map to 21011 and perhaps something else.

Reply:
Hmm, I see. It's up to you I am fine with both, even though the simple localhost is more readable.